### ISSUE TITLE
Docker API version

### ISSUE THREAD

**Issue by blabla1337** [lucasteligioridis/raspbernetes#6]
Hi @lucasteligioridis,

Me again, so for the installation part its fixed with the ```export DOCKER_API_VERSION=1.39``` pinning.
But now when you are on the boxes and try to run docker you get the same error and issue.
```
root@rpi-kube-master-01:/home/pi# docker ps
Error response from daemon: client version 1.40 is too new. Maximum supported API version is 1.39
```
after the export cmd the docker cmd works again.

We should add it in the users env by default, what do you think?

Kind regards,
Glenn

**Comment by lucasteligioridis** [comment]
I just did a quick search, can't seem to find a way to pin this version on install somewhere?

I wonder if something can be pinned?

I didn't receive this when I set this up, so just wondering what was not immutable. 

I haven't got my Pi's setup at the moment, can you have a try with maybe switching the docker version? https://github.com/lucasteligioridis/raspbernetes/blob/master/raspbernetes/install/docker.sh#L4

According to this: https://github.com/kubernetes-sigs/kubespray/issues/6160

Should be an easy fix with a change to the version of `docker-ce=5:19.03.8~3-0~ubuntu-bionic`. Will have to find the corresponding raspbian version, but I reckon try that first. It feels cleaner and then we can remove the exported DOCKER_API_VERSION variable from the configuration. 

Thoughts?

**Comment by blabla1337** [comment]
Hmm its not available?
root@rpi-kube-worker-02:/home/pi# sudo apt-get install docker-ce=5:19.03.8~3-0~ubuntu-bionic docker-ce-cli=5:19.03.8~3-0~ubuntu-bionic containerd.io
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Version '5:19.03.8~3-0~ubuntu-bionic' for 'docker-ce' was not found

**Comment by lucasteligioridis** [comment]
Yep, like i said you'll have to find the raspbian version of this. The above was just an example of ubuntu. 

You can try something like the below and grep the results.
```
apt-cache search docker-ce
```

**Comment by lucasteligioridis** [comment]
Did you get any luck with this?

**Comment by blabla1337** [comment]
I ran it and it wants to install the  ```5:19.03.9~3-0~raspbian-stretch```

I tried it on one Pi and seems to work, the Docker API error is gone.

**Comment by lucasteligioridis** [comment]
Awesome stuff! I reckon that should be pinned on the apt install and we won’t need the DOCKER_API variable anymore. 

Did you have that set as well?

**Comment by blabla1337** [comment]
Yeah so before I couldn't run it and got the API error, then installed this version and after it the docker api / commands are working correctly 

Indeed we can undo my commit and pin that version and should be good :)

**Comment by lucasteligioridis** [comment]
Mind please raising a PR with exactly what you did in that install file? And undoing your commit.

Otherwise I’m happy to!

**Comment by blabla1337** [comment]
Done, its here:
https://github.com/lucasteligioridis/raspbernetes/pull/7

**Comment by lucasteligioridis** [comment]
Thanks for all this mate! Awesome stuff!
Fixed in #7.
